Framework/API/src/IFunction.cpp

```diff
@@ -61,13 +62,31 @@ struct TieNode {
            other.right.end();
   }
 };
+const std::vector<std::string> EXCLUDEUSAGE = {"CompositeFunction"};
 } // namespace
+/**
+ * Constructor
+ */
+IFunction ::IFunction()
+    : m_isParallel(false), m_handler(nullptr), m_chiSquared(0.0) {}
 
 /**
  * Destructor
  */
 IFunction::~IFunction() { m_attrs.clear(); }
 
+/**
+Registers the usage of the function with the UsageService
+ */
+void IFunction::registerFunctionUsage() {
+  if (std::find(EXCLUDEUSAGE.cbegin(), EXCLUDEUSAGE.cend(), name()) ==
```

Comment:
I was just checking the [Algorithm implementation](https://github.com/mantidproject/mantid/blob/c90097a71777dc2ba5c195b3530f65a77ef95824/Framework/API/src/Algorithm.cpp#L1834) and there it also has a check to see if the service is enabled before posting. I'm not sure why this check isn't inside the `registerFeatureUsage` function rather than forcing everyone to do it.